### ISSUE TITLE
Snapshot generator calculates invalid max cardinality for extensions

### ISSUE THREAD

**Issue by Rob5045** [FirelyTeam/firely-net-sdk#1981]
A bug has been reported for Forge:
https://firely.atlassian.net/browse/FOR-611

When adding a new extension to a profile, save it and reload it, the max cardinality is set to "*" while "1" is expected.

I have created a unit test in this branch (based on 3.8.0 R4 release) to reproduce the problem:
https://github.com/FirelyTeam/firely-net-sdk/tree/forge/FOR-611

The expected behavior is that the BaseElement and derived element contains the correct max cardinality.

FHIR Version: R4 (may apply to other versions as well)
SDK Version: 3.6.0 and newer (works as expected in version 3.4.0!)

**Comment by ewoutkramer** [comment]
Unfortunately, the SDK team does not maintain the snapshot generator anymore since 2022-02-20. We are contemplating restoring the original situation where the Forge team was responsible for its maintenance 🤣

**Comment by Rob5045** [comment]
As of 2022-02-20 I have secretly the Simplifier team!

**Comment by Rob5045** [comment]
As per response from Ardon this problem is a showstopper for the upcoming Forge 28.0 release.

**Comment by ewoutkramer** [comment]
Since the show must go on, I'll flag this issue.

**Comment by marcovisserFurore** [comment]
I am browsing the release notes between 3.4 and 3.6, but I have not seen a smoking gun yet.  But we will find it :)

**Comment by ewoutkramer** [comment]
> When adding a new extension to a profile, save it and reload it, the max cardinality is set to "*" while "1" is expected.

Why is the "1" expected? I followed the link to the issue in Forge, but that only contains a screenshot, so it was not clear to me what is actually wrong with it.

**Comment by Rob5045** [comment]
The extension has a cardinality of 0..1 so when you use it in a profile the "base element" should have a cardinality of 0..1 as well. However after the snapshot generator does it's thing it has a cardinalty of 0..* (see the unit test in the branch https://github.com/FirelyTeam/firely-net-sdk/tree/forge/FOR-611). When you run this unit test from the 3.4.0 branch it is green. When run from 3.8.0 it is red.

**Comment by marcovisserFurore** [comment]
I found the culprit. The PR #1826 changed this behavior and @ewoutkramer already warned for that: https://github.com/FirelyTeam/firely-net-sdk/pull/1826#discussion_r678056878.

The proposed solution (the higher one of the `min` and the lower one of the `max`) is working for the `max`, but not for the `min` . I am looking for a better solution, but the good news is that we found the place to fix it.
